src/shared/lib/LazyMobXCache.ts

```diff
@@ -135,17 +147,30 @@ export default class LazyMobXCache<Data, Query> {
         this.updateCache(toMerge);
     }
 
-    private putData(queries:Query[], data:Data[]) {
-        const toMerge:Cache<Data> = {};
+    private putData(queries:Query[], data:FetchResult<Data, Metadata>) {
+        const toMerge:Cache<Data, Metadata> = {};
         const keyHasData:{[key:string]:boolean} = {};
 
-        for (const datum of data) {
-            const datumKey = this.dataToKey(datum);
-            toMerge[datumKey] = {
-                status: "complete",
-                data: datum
-            };
-            keyHasData[datumKey] = true;
+        for (const dataElt of data) {
```

Comment:
this could use some explanatory comment.  if i understand correctly, the presence of meta means that dataElt.data is a collection whose members need to be merged (and given ref to meta) wheareas without meta, only the parent would be merged?